Tests/FluentAssertions.Shared.Specs/MethodInfoSelectorAssertionSpecs.cs

```diff
@@ -141,9 +141,104 @@ public void When_asserting_methods_are_decorated_with_attribute_but_they_are_not
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldNotThrow();
+        }
+
+        [TestMethod]
+        public void When_asserting_methods_are_not_decorated_with_attribute_but_they_are_it_should_throw()
+        {
+            //-------------------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-------------------------------------------------------------------------------------------------------------------
+            MethodInfoSelector methodSelector =
+                new MethodInfoSelector(typeof(ClassWithAllMethodsDecoratedWithDummyAttribute))
+                    .ThatArePublicOrInternal;
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () =>
+                methodSelector.Should().NotBeDecoratedWith<DummyMethodAttribute>();
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldThrow<AssertFailedException>();
+        }
+
+        [TestMethod]
+        public void When_asserting_methods_are_not_decorated_with_attribute_but_they_are_it_should_throw_with_descriptive_message()
+        {
+            //-------------------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-------------------------------------------------------------------------------------------------------------------
+            var methodSelector = new MethodInfoSelector(typeof(ClassWithAllMethodsDecoratedWithDummyAttribute));
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () =>
+                methodSelector.Should().NotBeDecoratedWith<DummyMethodAttribute>("because we want to test the error {0}", "message");
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldThrow<AssertFailedException>()
+                .WithMessage("Expected no selected methods to be decorated with" +
```

Comment:
The `WithMessage` takes a wildcard pattern, so I would recommend verifying only the relevant parts of the message (as explained in [this blog post](http://www.continuousimprover.com/2015/11/12-tips-to-write-unit-tests-that-dont.html)). IMO, you don't want a failing test just because the wording of the message has changed.  Same for the other specs.